### PR TITLE
Add application overlay for notebook-controller 

### LINKED ISSUE #450: add application overlay for notebook-controller
required for upgrade and delete

### PR DESCRIPTION
**Pull request by kkasravi**
**Which issue is resolved by this Pull Request:**
Resolves #450

**Description of your changes:**
standardized commonLabels, grouped resources by kind

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate`
    3. `make test`

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/451)

### CODE REVIEW

**Comment by jlewi** [comment]
@kkasravi I would expect this PR to only contain changes to the Application resource and the kustomization.yaml file

But for katib it looks like there are other changes to the kustomize manifest.

**Comment by richardsliu** [comment]
Maybe just separate out notebook-controller and katib?

**Comment by kkasravi** [comment]
> Maybe just separate out notebook-controller and katib?

sure

**Comment by jlewi** [comment]
@kkasravi nit can you please revert any tests not related to the changed manifests.

**Comment by richardsliu** [comment]
@jlewi I suspect those files were autoformatted by Go.

**Comment by kkasravi** [comment]
i'll revert the go tests that were picked up

File: jupyter/notebook-controller/overlays/application/application.yaml [hunk 1]
```diff
@@ -6,11 +6,11 @@ spec:
   selector:
     matchLabels:
       app.kubernetes.io/name: notebook-controller
-      app.kubernetes.io/instance: notebook-controller
+      app.kubernetes.io/instance: notebook-controller-v0.6.2
```
Comment (jlewi) [inline jupyter/notebook-controller/overlays/application/application.yaml]: Do you want to set the version to v0.7.0?

File: jupyter/notebook-controller/overlays/application/kustomization.yaml [hunk 2]
```diff
@@ -6,8 +6,8 @@ resources:
 - application.yaml
 commonLabels:
   app.kubernetes.io/name: notebook-controller
-  app.kubernetes.io/instance: notebook-controller
+  app.kubernetes.io/instance: notebook-controller-v0.6.2
```
Comment (jlewi) [inline jupyter/notebook-controller/overlays/application/kustomization.yaml]: v0.7.0

File: katib/katib-controller/base/katib-controller-rbac.yaml [hunk 3]
```diff
@@ -84,63 +84,3 @@ roleRef:
 subjects:
 - kind: ServiceAccount
   name: katib-controller
-
----
```
Comment (jlewi) [inline katib/katib-controller/base/katib-controller-rbac.yaml]: Why did you delete this?

File: katib/katib-controller/base/katib-manager-deployment.yaml [hunk 4]
```diff
@@ -43,5 +43,3 @@ spec:
           exec:
             command: ["/bin/grpc_health_probe", "-addr=:6789"]
           initialDelaySeconds: 10
-          periodSeconds: 60
```
Comment (jlewi) [inline katib/katib-controller/base/katib-manager-deployment.yaml]: Why did you change this?

Comment (kkasravi) [reply katib/katib-controller/base/katib-manager-deployment.yaml]: merge error, i'll revert

File: katib/katib-controller/base/katib-controller-rbac.yaml (hunk 3 above)
Comment (kkasravi) [reply katib/katib-controller/base/katib-controller-rbac.yaml]: mergo error, i'll revert

**Comment by kkasravi** [comment]
@richardsliu @jlewi 
this is ready for merge
